### PR TITLE
fix(centreonACL): return all services linked to a servicegroup

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

The issue here was that when linking service templates to define a servicegroup, those where not "resolved" which lead to the non-consistent listing in the Reporting -> Dashboard -> Service Groups feature.

Using the centreon_storage database to get the information, where the services_servicegroups table provides "resolved" dependencies of possible components of the servicegroup which can be:
- simple services
- service templates
- hostgroup services

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [x] 19.04.x
- [x] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Create a service group using all the possible components listed above.
- Generate and push the configuration
- Go to the reporting page of the service group.

The listing should display all the services linked to the servicegroup.

**IMPORTANT**

The test should also be done using an user with ACL on ressources.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/class/centreonACL.class.php [hunk 1]
```diff
@@ -2058,24 +2064,21 @@ public function getServiceServiceGroupAclConf($sg_id, $broker = null, $options =
...
-            . "UNION "
-            . "SELECT " . $request['fields'] . " "
-            . "FROM servicegroup, servicegroup_relation, hostgroup_relation, service, host " . $from_acl . " "
-            . "WHERE servicegroup.sg_id = '" . CentreonDB::escape($sg_id) . "' "
-            . "AND servicegroup.sg_id = servicegroup_relation.servicegroup_sg_id "
-            . "AND servicegroup_relation.hostgroup_hg_id = hostgroup_relation.hostgroup_hg_id "
-            . "AND hostgroup_relation.host_host_id = host.host_id "
-            . "AND servicegroup_relation.service_service_id = service.service_id "
+            . "FROM " .$db_name_acl . ".services_servicegroups, service, host" . $from_acl . " "
+            . "WHERE servicegroup_id = " . CentreonDB::escape($sg_id) . " "
```
Comment (sc979) [inline www/class/centreonACL.class.php]: please could you use camelCase on $sg_id and could you bind it

File: www/class/centreonACL.class.php [hunk 2]
```diff
@@ -2058,24 +2064,21 @@ public function getServiceServiceGroupAclConf($sg_id, $broker = null, $options =
...
-            . $where_acl . " "
-            . "UNION "
-            . "SELECT " . $request['fields'] . " "
-            . "FROM servicegroup, servicegroup_relation, hostgroup_relation, service, host " . $from_acl . " "
-            . "WHERE servicegroup.sg_id = '" . CentreonDB::escape($sg_id) . "' "
-            . "AND servicegroup.sg_id = servicegroup_relation.servicegroup_sg_id "
-            . "AND servicegroup_relation.hostgroup_hg_id = hostgroup_relation.hostgroup_hg_id "
-            . "AND hostgroup_relation.host_host_id = host.host_id "
-            . "AND servicegroup_relation.service_service_id = service.service_id "
+            . "FROM " .$db_name_acl . ".services_servicegroups, service, host" . $from_acl . " "
```
Comment (sc979) [inline www/class/centreonACL.class.php]:
```suggestion
            . "FROM " . $db_name_acl . ".services_servicegroups, service, host" . $from_acl . " "
```

File: www/class/centreonACL.class.php [hunk 3]
```diff
@@ -2058,24 +2064,21 @@ public function getServiceServiceGroupAclConf($sg_id, $broker = null, $options =
...
+         *     - simple services
+         *     - service templates
+         *     - hostgroup services
+         */
         $query = $request['select'] . $request['simpleFields'] . " "
             . "FROM ( "
             . "SELECT " . $request['fields'] . " "
-            . "FROM servicegroup, servicegroup_relation, service, host " . $from_acl . " "
-            . "WHERE servicegroup.sg_id = '" . CentreonDB::escape($sg_id) . "' "
-            . "AND service.service_activate='1' AND host.host_activate='1' "
```
Comment (sc979) [inline www/class/centreonACL.class.php]: are you sure that you want results of disabled resources ?

Comment (adr-mo) [reply www/class/centreonACL.class.php]: `AND service.service_activate='1' AND host.host_activate='1'` is this condition not enough to only get enabled ressources ?

If your question is "Do I want the disabled ressources ?" The answer is no :smiley:

Comment (sc979) [reply www/class/centreonACL.class.php]: so you should restore this line ;)

File: www/class/centreonACL.class.php (hunk 1 above)
Comment (adr-mo) [reply www/class/centreonACL.class.php]: Should be good now :)

File: www/class/centreonACL.class.php (hunk 2 above)
Comment (adr-mo) [reply www/class/centreonACL.class.php]: Should be good now :)

File: www/class/centreonACL.class.php (hunk 3 above)
Comment (adr-mo) [reply www/class/centreonACL.class.php]: Should be good now :)

**Comment by adr-mo** [comment]
Feedbacks taken into account ! Thanks @sc979 for feedback and help :+1:

File: www/class/centreonACL.class.php [hunk 4]
```diff
@@ -2058,24 +2064,29 @@ public function getServiceServiceGroupAclConf($sg_id, $broker = null, $options =
                 . "AND $db_name_acl.centreon_acl.host_id = host.host_id "
                 . "AND $db_name_acl.centreon_acl.service_id = service.service_id ";
         }
+
+        // Making sure that the id provided is a real int
+        $option = array (
+            'default' => 0
+        );
```
Comment (kduret) [inline www/class/centreonACL.class.php]:
```suggestion
        $option = ['default' => 0];
```

**Pull request merged**